pkg/controller/controller.go

```diff
@@ -1165,3 +1159,50 @@ func (m *monitoringPackage) CreateVmiSecrets(binding *types.SyntheticBinding, se
 func (m *monitoringPackage) DeletePomPusher(binding string, helper v8outil.DeploymentHelper) error {
 	return v8omonitoring.DeletePomPusher(binding, helper)
 }
+
+// MCRegistrationSecret - the name of the secret that contains the cluster registration information
+const MCRegistrationSecret = "verrazzano-cluster-registration"
+
+// ClusterNameData - the field name in MCRegistrationSecret that contains this managed cluster's name
+const ClusterNameData = "managed-cluster-name"
+
+// ElasticsearchSecretName - the name of the secret in the Verrazzano System namespace,
+// that contains credentials and other details for for the admin cluster's Elasticsearch endpoint
+const ElasticsearchSecretName = "verrazzano-cluster-elasticsearch"
+
+// ElasticsearchURLData - the field name in ElasticsearchSecret that contains the admin cluster's
+// Elasticsearch endpoint's URL
+const ElasticsearchURLData = "url"
+
+// ElasticsearchUsernameData - the field name in ElasticsearchSecret that contains the admin
+// cluster's Elasticsearch username
+const ElasticsearchUsernameData = "username"
+
+// ElasticsearchPasswordData - the field name in ElasticsearchSecret that contains the admin
+// cluster's Elasticsearch password
+const ElasticsearchPasswordData = "password"
+
+// ElasticsearchCABundleData - the field name in ElasticsearchSecret that contains the admin
+// cluster's Elasticsearch ca-bundle
+const ElasticsearchCABundleData = "ca-bundle"
+
+// getManagedClusterName returns the cluster name for a managed cluster, empty string otherwise
+func (c *Controller) getClusterInfo() v8omonitoring.ClusterInfo {
+	managedClusterName := ""
+	clusterSecret, err := c.secrets.Get(MCRegistrationSecret)
+	if err == nil {
```

Comment:
handle err != nil

Reply:
when err != nil, ClusterInfo is constructed with empty string ManagedClusterName.